apps/sms/js/subject_composer.js

```diff
@@ -106,7 +113,8 @@
 
       // methods
       updateValue: updateValue.bind(this),
-      updatePlaceholder: updatePlaceholder.bind(this)
+      updatePlaceholder: updatePlaceholder.bind(this),
+      updateDirection: updateDirection.bind(this)
```

Comment:
input is only defined in private set. I think it's the only way to access private propery/function in subject_component